pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -681,9 +721,12 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 				}
 			}
 			if !fullWildcardFound {
-				match = &listener.FilterChainMatch{SniDomains: chain.sniHosts}
+				match.SniDomains = chain.sniHosts
 			}
 		}
+		if reflect.DeepEqual(*match, listener.FilterChainMatch{}) {
+			match = nil
+		}
```

Comment:
isn't it easier to simply assign the filter chain match only if transportProtocol is set?

Reply:
No. `sniHost` can also make filter chain match non-empty. But based on discussion with @mandarjog we don't want to add an empty filter chain match if neither sni or multiplexing requires.